boards/nucleo-l152re/include/periph_conf.h

```diff
@@ -226,6 +226,11 @@ static const spi_conf_t spi_config[] = {
         .miso_pin = GPIO_PIN(PORT_A, 6),
         .sclk_pin = GPIO_PIN(PORT_A, 5),
         .cs_pin   = GPIO_UNDEF,
+#ifdef MODULE_PERIPH_SPI_GPIO_MODE
```

Comment:
I think by default this should be set as it is now, so:

```
        .mosi_pin_mode = (GPIO_OUT),
        .miso_pin_mode = (GPIO_IN),
        .sclk_pin_mode = (GPIO_OUT),
```

But in concerned driver add `FEATURES_OPTIONAL = periph_spi_gpio_mode`. And change the gpio mode accordingly.

IMO having it as a default configuration for a board isn't the best. If an spi device uses CPOL=1 you would want to have a PU and not PD. A driver could have pull ups or pull down on the board witch could affect everything, for example some devices can operate ass SPI/I2C and would then have pull ups, if you pull it down by default the current leak will be higher.

I think you could implement this in a way where a device driver can modify the default mosi/miso/sclk pin mode. I'm not sure how this would be implemented... I can think of using CFLAGS (not ideal) or adding an spi function that changes the pins modes having a separate structure for this?

```
void spi_init_pins_mode(spi_t bus, spi_pin_mode_t mode)
{
    spi_config[bus].mosi_pin_mode = mode.mosi;
    spi_config[bus].miso_pin_mode = mode.miso;
    spi_config[bus].sclk_pin_mode = mode.sclk;
}

// or just

void spi_init_pins_mode(spi_t bus, spi_pin_mode_t mode)
{
    gpio_init(spi_config[bus].mosi_pin, mode.mosi);
    gpio_init(spi_config[bus].miso_pin, mode.miso);
    gpio_init(spi_config[bus].sclk_pin, mode.sclk);
}
```

There could still be conflicts between spi devices but that is up to the user to realize that using spi_devices with different inactive state is not a good idea.

What do you think?

Reply:
> What do you think?

I like the idea. I'll provide a patch to this PR.